components/CollectionForm.js

```diff
@@ -0,0 +1,115 @@
+import { useState } from "react";
+import styled from "styled-components";
+import Button from "@/components/Button";
+
+const StyledFormContainer = styled.div`
+  display: flex;
+  align-items: center;
+  flex-direction: column;
+  width: 100vw;
+  padding: 3em;
+`;
+
+const StyledForm = styled.form`
+  display: flex;
+  flex-direction: column;
+  width: 250px;
+  gap: 10px;
+`;
+
+const StyledColors = styled.button`
+  width: 24px;
+  height: 24px;
+  border-radius: 50px;
+  border: none;
+  cursor: pointer;
+  background-color: ${(prop) => prop.$inputColor};
+`;
+
+const StyledColorPicker = styled.div`
+  display: flex;
+  flex-direction: row;
+  align-items: center;
+  gap: 12px;
+`;
+
+const StyledButtonContainer = styled.div`
+  display: flex;
+  flex-direction: row;
+  justify-content: space-evenly;
+  margin-top: 50px;
+`;
+
+const StyledSubmitMessage = styled.p`
+  text-align: center;
+`;
+
+export default function CollectionForm({ onSubmitCollection, prevValues }) {
+  const [confirmMessage, setConfirmMessage] = useState("");
+  const [collectionColor, setCollectionColor] = useState(prevValues?.color);
+  const initialColors = ["#c28375", "#DAF7A6", "#75c297", "#7597c2", "#ab75c2"];
+
+  function handleCancel(event) {
+    event.preventDefault();
+    event.target.form.reset(); // reset the form
+    setCollectionColor(prevValues?.color || ""); // Reset color to previous value
+  }
+
+  return (
+    <>
+      <StyledFormContainer>
+        <StyledForm
+          onSubmit={(e) => {
+            onSubmitCollection(e, prevValues?.id), setConfirmMessage(true);
+          }}
+        >
+          <label htmlFor="title" required>
+            Title:
+          </label>
+          <input
+            type="text"
+            id="title"
+            name="title"
+            required
+            defaultValue={prevValues?.title || ""}
+          />
+
+          <label htmlFor="colorPicker">Color:</label>
+          <StyledColorPicker>
+            {initialColors.map((color, index) => (
+              <StyledColors
+                type="button"
+                aria-label="color-picker"
+                $inputColor={color}
+                key={index}
```

Comment:
please don't use the index of the array as the key. In fact it is the same as giving no key at all, since the index is used automatically if no key is provided. The value of the key must be a) unique and b) strickly coupled with the data element.